tests/driver_lis2dh12/main.c

```diff
@@ -48,25 +93,69 @@ int main(void)
         return 1;
     }
 
-    xtimer_ticks32_t last_wakeup = xtimer_now();
+    /* enable interrupt Pins */
+#ifdef LIS2DH12_INT_PIN1
+    if (gpio_init_int(LIS2DH12_INT_PIN1,GPIO_IN, GPIO_RISING,lis2dh12_int_cb,"INT_1") == -1)
```

Comment:
I'd move that into `lis2dh12_set_int()`.
Also don't hard-code `LIS2DH12_INT_PIN1` but rather add an entry to `lis2dh12_params_t` - then you don't have to specify the pin but can just use the `int_line` parameter.

It also makes it possible to have more than one instance of the lis2dh12 driver running, we should not regress this if we want to get this merged.